plugins/hu.bme.mit.massif.simulink.api/src/hu/bme/mit/massif/simulink/api/Importer.java

```diff
@@ -794,12 +803,15 @@ private void createBlocksFromTopLevel(SimulinkModel simulinkModel) throws Simuli
 
         // Get the handle of the model itself
         MatlabCommand getModelHandle = commandFactory.getParam().addParam(modelFQN).addParam("Handle");
-        Double modelHandle = Handle.getHandleData(getModelHandle.execute());
+        IVisitableMatlabData modelHandle = getModelHandle.execute();
+        Double modelHandleValue = Handle.getHandleData(modelHandle);
 
+        blockHandleCache.put(modelFQN, (Handle) modelHandle);
```

Comment:
Is it really worth it to have a cache instead of passing the handle through the call hierarchy (maybe in the same DTO as suggested below)